### DISCUSSION TITLE
Question: Can I use native lazy loading with <Image>?

### DISCUSSION THREAD

**Discussion by MaxMEllon** [vercel/next.js#18526] [Help]
# Question
refs: #16832

Can I use native lazy loading with `next/image` ?
Canary implementation is always lazy loading by `IntersectionObserver` (if able to use IntersectionObserver) . [refs](https://github.com/vercel/next.js/blob/canary/packages/next/client/image.tsx#L235-L381)

- rewrite and check loading attr in chrome. [source](https://github.com/MaxMEllon/next.js/pull/1/files)

<img width="1351" alt="スクリーンショット 2020-10-28 10 25 03" src="https://user-images.githubusercontent.com/9594376/97382382-a1387380-190e-11eb-943e-01bd4377ea2c.png">

> "auto" is default value in chrome, out of the whatwg spec.

I think the best implementation is switching strategy native or not with feature detection like as `'loading' in HTMLImageElement.prototype`.

What about do you think?
Why dont use native lazy loading?

**Comment by jacobrask** [comment]
On the server, where the HTML is generated, you won’t know if native lazy loading is supported.

It would be nice to be able to opt in to it though.

**Reply by MaxMEllon** [reply]
I think can includes any string as attribute in html tag ([whatwg 3.2.4.1](https://www.w3.org/TR/2011/WD-html5-20110525/elements.html#attributes)). So we can put the `loading` attribute to html in always (will be avoided the loading attribute in legacy browser and  images will load as usual).
Then, we can switch strategies based on feature detection in client side.


```
3.2.4.1 Attributes
Except where otherwise specified, attributes on HTML elements may have any string value, 
including the empty string. Except where explicitly stated, there is no restriction on what text can be specified in such attributes.
```

Actually, [gatsbyjs/gatsby](https://www.gatsbyjs.com/plugins/gatsby-remark-images-native-lazy-load/#what-does-this-plugin-do) is putting the `loading` atttribute in anytime. (but does not switch strategies)
